filip/clients/ngsi_v2/cb.py

```diff
@@ -1386,6 +1390,137 @@ def post_command(self,
+            if not str(ex)[0:36] == "404 Client Error: Not Found for url:":
+                raise
+            return False
+        
+        return True
+
+    def patch_entity(self,
+                     entity: ContextEntity,
+                     old_entity: Optional[ContextEntity] = None) -> None:
+        """
+           Takes a given entity and updates the state in the CB to match it.
+           Args:
+               entity: Entity to update
+               old_entity: OPTIONAL, if given only the differences between the
+                           old_entity and entity are updated in the CB.
+                           Other changes made to the entity in CB, can be kept.
+           Returns:
+               None
+        """
+
+        new_entity = entity
+
+        if old_entity is None:
+            # If no old entity_was provided we use the current state to compare
+            # the entity to
+            if self.does_entity_exists(entity_id=new_entity.id,
+                                       entity_type=new_entity.type):
+                old_entity = self.get_entity(entity_id=new_entity.id,
+                                             entity_type=new_entity.type)
+            else:
+                # the entity is new, post and finish
+                self.post_entity(new_entity, update=False)
+                return
+
+        else:
+            # An old_entity was provided
+            # check if the old_entity (still) exists else recall methode
+            # and discard old_entity
+            if not self.does_entity_exists(entity_id=old_entity.id,
+                                           entity_type=old_entity.type):
+                self.patch_entity(new_entity)
+                return
+
+            # if type or id was changed, the old_entity needs to be deleted
+            # and the new_entity created
+            # In this case we will loose the current state of the entity
+            if old_entity.id != new_entity.id or \
+                    old_entity.type != new_entity.type:
+                self.delete_entity(entity_id=old_entity.id,
+                                   entity_type=old_entity.type)
+
+                if not self.does_entity_exists(entity_id=new_entity.id,
+                                               entity_type=new_entity.type):
+                    self.post_entity(entity=new_entity, update=False)
+                    return
+
+        # At this point we know that we need to patch only the attributes of
+        # the entity
+        # Check the differences between the attributes of old and new entity
+        # Delete the removed attributes, create the news,
+        # and update the existing if necessary
+        old_attributes = old_entity.get_properties()
+        old_attributes.extend(old_entity.get_relationships())
+
+        new_attributes = new_entity.get_properties()
+        new_attributes.extend(new_entity.get_relationships())
+
+        # Manage attributes that existed before
```

Comment:
actually this applies to all attributes that are provided via registration. Think that you finally have to check if there are corresponding registrations. https://fiware-tutorials.readthedocs.io/en/latest/context-providers.html check for the registrations and check them if they match the entity attributes. All matching attributes do probably come from registrations